transaction/core/src/input_rules.rs

```diff
@@ -53,6 +69,103 @@ impl InputRules {
                 return Err(InputRuleError::MissingRequiredOutput);
             }
         }
+        // Partial-Fill rules verification (MCIP #42)
+        if let Some(fractional_change) = self.fractional_change.as_ref() {
+            // There is a fractional change output. Let's try to unblind its amount.
+            let fractional_change_amount = fractional_change.reveal_amount()?;
+            // Let's check if there is a corresponding real change output.
+            let real_change = tx
+                .prefix
+                .outputs
+                .iter()
+                .find(|x| {
+                    x.public_key == fractional_change.tx_out.public_key
+                        && x.target_key == fractional_change.tx_out.target_key
```

Comment:
Same comment as below -- we really should have a function on `TxOut`'s that checks if they are equal everywhere excepting the masked amount, and call that function here.

Reply:
You can have a helper function that clones one of the sides of the equation and override its masked amount with the other side, that ensures everything gets compared other than the masked amount.
A bit slower but probably doesn't matter